### DISCUSSION TITLE
Hosting for Taskwarrior - Options?

### DISCUSSION THREAD

**Discussion by ersanchez** [GothenburgBitFactory/taskwarrior#3066] [Q&A]
Does anyone know of any free or paid options for hosting taskwarrior?

I've tried multiple times over the years to self-host it but, sadly, I have never been successful.

**Comment by djmitche** [comment]
Until recently, freecinc.com was an answer to this, but I shut it down a week or so ago :(

@hickey has built https://gitlab.com/wt0f/taskd/, which may help standing up your own version inside Docker.

**Comment by Arwalk** [comment]
I've successfully hosted my taskwarrior server using https://github.com/j6s/docker-taskwarrior-taskserver .

Setup is trivial, highly recommend.

**Comment by traycerb** [comment]
for those more sophisticated, esp in Docker, I suppose it is quite easy, but I too found it quite challenging.

I ended up getting [this version](https://github.com/ogarcia/docker-taskd) working on my Synology NAS.  Spinning it up to work locally was no problem, but getting the certificates to use remotely was challenging.  

But now it works great.  I will try to do a write up if I can.  Part of the challenge I think was doing things on the NAS, with it's own quirks and brand of linux.

**Comment by ruapho** [comment]
I am using [inthe.am](https://inthe.am/). Also offering their [source](https://github.com/coddingtonbear/inthe.am) for self-hosting. Looks quite complicated to self-host, but the support docker.

**Reply by slarrain** [reply]
Thanks for suggesting inthe.am. Tried it and its working great. Super simple to set up.

**Comment by kirkster96** [comment]
> Does anyone know of any free or paid options for hosting taskwarrior?
> 
> I've tried multiple times over the years to self-host it but, sadly, I have never been successful.

In order to help those looking for another solution, I made a tutorial for how to deploy taskd on your own Azure VM:

https://youtu.be/6Ci_JyvVIaI
